### ISSUE TITLE
FasterTransformer: Start to forward terminate called after throwing an instance of 'std::out_of_range'

### ISSUE THREAD

**Issue by eycheung** [triton-inference-server/server#5396]
**Description**
When calling `InferenceServerClient.infer` with my supplied T5 Pytorch model using `FasterTransformer`, I get the `std::out_of_range` error.

Full stack trace:
```
I0222 21:21:52.596277 15159 libfastertransformer.cc:834] Start to forward
terminate called after throwing an instance of 'std::out_of_range'
  what():  _Map_base::at
Signal (6) received.
 0# 0x0000561BF3F2D549 in /opt/tritonserver/bin/tritonserver
 1# 0x00007FA68048E090 in /usr/lib/x86_64-linux-gnu/libc.so.6
 2# gsignal in /usr/lib/x86_64-linux-gnu/libc.so.6
 3# abort in /usr/lib/x86_64-linux-gnu/libc.so.6
 4# 0x00007FA680847911 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 5# 0x00007FA68085338C in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 6# 0x00007FA6808533F7 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 7# 0x00007FA6808536A9 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 8# std::__throw_out_of_range(char const*) in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 9# 0x00007FA511BC7FFC in /opt/tritonserver/backends/fastertransformer/libtransformer-shared.so
10# T5TritonModelInstance<__half>::forward(std::shared_ptr<std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, triton::Tensor, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, triton::Tensor> > > >) in /opt/tritonserver/backends/fastertransformer/libtransformer-shared.so
11# 0x00007FA5B0F5E0CA in /opt/tritonserver/backends/fastertransformer/libtriton_fastertransformer.so
12# 0x00007FA68087FDE4 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
13# 0x00007FA680CFC609 in /usr/lib/x86_64-linux-gnu/libpthread.so.0
14# clone in /usr/lib/x86_64-linux-gnu/libc.so.6
```

**Triton Information**
22.03

Are you using the Triton container or did you build it yourself?
Using a Triton container. Used the same containers as `fastertransformer_backend/notebooks/GPT-J_and_T5_inference.ipynb`.

**To Reproduce**

1. Start with a pretrained `T5ForConditionalGeneration` model as input (from [Huggingface](https://huggingface.co/docs/transformers/v4.26.1/en/model_doc/t5#transformers.T5ForConditionalGeneration)).

Modify this slightly to just return the logits from the forward pass.

```
from transformers import AutoConfig
config = AutoConfig.from_pretrained(PATH_TO_MODEL_FILES)

class AugmentedModel(T5ForConditionalGeneration):

    def forward(self, *args, **kwargs):
        return super().forward(*args, **kwargs).logits

augmented_model = AugmentedModel(config).eval().to(device).half()
augmented_model.save_pretrained(PATH_TO_ADJUSTED_MODEL)

augmented_model.forward(input_ids, mask, decoder_input_ids) # <-- this works.
```

2. Run the t5 checkpointer converted
```
python3 FasterTransformer/examples/pytorch/t5/utils/huggingface_t5_ckpt_convert.py \
        -i PATH_TO_ADJUSTED_MODEL/ \
        -o ./models/t5/ \
        -i_g 1 \
        -weight_data_type fp16
``` 

3. Create the gemm file
```
!CUDA_VISIBLE_DEVICES=0 ./FasterTransformer/build/bin/t5_gemm 1 1 32 512 8 64 1024 512 8 64 1024 32100 1 1 1 0
```

4. Create the `config.pbtxt` file

```
name: "fastertransformer"
backend: "fastertransformer"
default_model_filename: "t5"
max_batch_size: 128
input [
  {
    name: "input_ids"
    data_type: TYPE_INT32
    dims: [ -1 ]
  },
  {
    name: "attention_mask"
    data_type: TYPE_INT32
    dims: [ -1 ]
  },
  {
    name: "decoder_input_ids"
    data_type: TYPE_INT32
    dims: [ -1 ]
  }
]
output [
  {
    name: "logits"
    data_type: TYPE_FP16
    dims: [ -1, 32100 ]
  }
]
instance_group [
  {
    count: 1
    kind : KIND_CPU
  }
]
parameters {
  key: "tensor_para_size"
  value: {
    string_value: "1"
  }
}
parameters {
  key: "pipeline_para_size"
  value: {
    string_value: "1"
  }
}
parameters {
  key: "is_half"
  value: {
    string_value: "1"
  }
}
parameters {
  key: "enable_custom_all_reduce"
  value: {
    string_value: "0"
  }
}
parameters {
  key: "model_type"
  value: {
    string_value: "T5"
  }
}
parameters {
  key: "model_checkpoint_path"
  value: {
    string_value: "/ft_workspace/models/t5/1-gpu"
  }
}
```
5. Start the TRITON server and make an inference request.
```
triton_client.infer(model_name="fastertransformer",
                                    inputs=triton_inputs,
                                    outputs=triton_outputs,
                                    headers={'test': '1'})
```

**Expected behavior**
Expect that the inference should run without issue.

Thanks in advance. Any guidance on how to debug this issue would be really helpful!

**Comment by rmccorm4** [comment]
Hi @eycheung, since this is FasterTransformer specific, please open an issue in the fastertransformer_backend issues page: https://github.com/NVIDIA/FasterTransformer/issues
